experiments/hello-network-socket-ping/local_pong.c

```diff
@@ -0,0 +1,212 @@
+
+// TODO consider rename to: prepare_for_bind_server
+static int
+prepare_for_bind(struct sockaddr_un* addr_ptr, const char* sock_path)
+{
+    char error_message[SIZE_ERROR_MESSAGE];
+    int32_t ret;
+
+    ret = 0;
+    memset(error_message, 0, SIZE_ERROR_MESSAGE);
+
+    if (strlen(sock_path) > sizeof(addr_ptr->sun_path) - 1) {
+        sscanf(
+                error_message,
+                "Socket path is too long (%d characters)",
+                strlen(sock_path));
+        error_message[255] = '\0';
+        print_error(error_message);
+        return -1;
+    }
+
+    // Ensure there is no pre-existing file with wanted name
+    ret = remove(sock_path);
+    if (-1 == ret && ENOENT != errno) {
+        perror("Failed to remove file");
+        sscanf(
+                error_message,
+                "Failed to remove file, using path: %s",
+                sock_path);
+        error_message[255] = '\0';
+        print_error(error_message);
+        return -1;
+    }
+
+    memset(addr_ptr, 0, sizeof(struct sockaddr_un));
+
+    addr_ptr->sun_family = AF_UNIX;
+
+    // Using sizeof the whole struct had two bytes (short) extra when compared
+    // to using sizeof sockaddr_un.sun_path. Checked from system header files.
+    strncpy(addr_ptr->sun_path, sock_path, sizeof(struct sockaddr_un) - 1);
+
+    return 0;
+}
+
+static int
+receive_message(
+        int32_t const * const sock_fd_ptr,
+        struct sockaddr_un* address_client_ptr,
+        char* out_message)
+{
+    socklen_t len;
+    ssize_t bytes_received;
+
+    char receive_buffer[SIZE_RECEIVE_BUFFER];
+
+    len = 0;
+    bytes_received = 0;
+    memset(receive_buffer, 0, SIZE_RECEIVE_BUFFER);
+
+    len = sizeof(struct sockaddr_un);
+    bytes_received = recvfrom(
+            *(sock_fd_ptr),
+            receive_buffer,
+            SIZE_RECEIVE_BUFFER,
+            0,
+            (struct sockaddr*) address_client_ptr,
+            &len);
+    if (-1 == bytes_received) {
+        perror("Failed to recvfrom");
+        return -1;
+    }
+
+#ifndef NDEBUG
+    (void) printf(
+            "Server received %ld bytes from: %s\n",
+            (long) bytes_received,
+            address_client_ptr->sun_path);
+#endif // NDEBUG
+
+    memcpy(out_message, receive_buffer, SIZE_RECEIVE_BUFFER);
+
+    return 0;
+}
+
+static int
+message_is_expected_message(char const * const message)
+{
+    const char* expected_message = "Ping";
+    int32_t expected_message_len = 4;
+    int32_t ret;
+
+    ret = 0;
+
+    ret = memcmp(message, expected_message, expected_message_len);
+
+    if (0 == ret) {
+        return 1;
+    }
+
+    return 0;
+}
+
+int
+main(void)
+{
+    const int32_t socket_protocol = 0;
+    const int32_t msg_size = 32;
+    const char* sock_path;
+
+    ssize_t bytecount;
+    int32_t latest_error;
+    int32_t sock_fd;
+    int32_t ret;
+
+    char message[SIZE_RECEIVE_BUFFER];
+
+    struct sockaddr_un address_server;
+    struct sockaddr_un address_client;
```

Comment:
Please zero these structs below.

Reply:
Done in: https://github.com/arbiter0xf/neotetris/pull/8/commits/4a6e67db7dd739e0d57c804041f6c0642d806b7d